charts/ChartConfig.tsx

```diff
@@ -270,6 +272,17 @@ export class ChartConfig {
                 this.props.dimensions = this.validDimensions
             }
         })
+
+        autorun(() => {
+            if (this.props.isExplorable) {
+                if (canBeExplorable(this.props)) {
+                    this.props.hasChartTab = true
+                    this.props.hasMapTab = true
```

Comment:
I thought a component was not supposed to update its own props?

Reply:
It's true that a component shouldn't _directly_ modify its props, and this codebase breaks that principle in many cases. It's basically built on breaking that principle – the chart config is often changed directly within components. See e.g. how the settings menu works: 

https://github.com/owid/owid-grapher/blob/b3b70fbb8eb7fad2d02935d416effe562a88e133/charts/Controls.tsx#L218-L220

Whether it's a good idea is debatable – it makes arbitrary config changes easy to make at the cost of making components harder to reason about. I would prefer the conventional React approach where mutating functions are explicitly passed, but that _would_ add a lot of boilerplate code.

Anyway, I was following the convention for applying "constraints" to config props; see the other autoruns above my additions. We can of course change that, how do you think we should ideally apply constraints to the config?

Reply:
OK. If this is already an established pattern let's just keep it for now. We can talk separately about whether we want to change to a different pattern, but it doesn't nee to hold up this PR.